db_stress_tool/db_stress_test_base.cc

```diff
@@ -633,6 +635,13 @@ void StressTest::OperateDb(ThreadState* thread) {
 
   const uint64_t ops_per_open = FLAGS_ops_per_thread / (FLAGS_reopen + 1);
 
+#ifdef FIND_BAD_RACE_CONDITION
+  ThreadStatusUtil::TEST_SetStateDelay(ThreadStatus::STATE_MUTEX_WAIT,
+                                       (thread->rand.Uniform(200)));
```

Comment:
Also as discussed, it seems better to me to do this at a per lock granularity rather than setting it per program or per thread. One reason is because this only enables it in the foreground so background work (which is where just about all the interesting DB mutex lock acquires/releases happen) isn't coercing context switching. Another reason is because combinations of long and short switches within a thread might be useful to test.

Reply:
> One reason is because this only enables it in the foreground so background work (which is where just about all the interesting DB mutex lock acquires/releases happen) isn't coercing context switching

Never mind about this reason. I thought ThreadStatusUtil would use thread-local state but it does not.

Reply:
>  Perhaps 0-10 milliseconds (integer values so no sleep can happen reasonably often) would be better.

Fixed

>Also as discussed, it seems better to me to do this at a per lock granularity

Fixed